src/Uri.php

```diff
@@ -611,7 +611,8 @@ private function filterHost($host)
             throw new \InvalidArgumentException('Host must be a string');
         }
 
-        return strtolower($host);
+        // strtolower() simply breaks internationalized domain names (depends on the locale selected)
+        return function_exists('mb_strtolower') ? mb_strtolower($host, 'UTF-8') : $host;
```

Comment:
Do we want to require the symfony polyfil or suggest ext-mb?

Reply:
Should we hard fail if the string contains non-asci characters and the mb extension or poylfil is not available?

Reply:
We can just replace this with: 

```
\strtr($host, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz')
```